docs/3.0.0-alpha.x/guides/deployment.md

```diff
@@ -867,14 +904,50 @@ Ensure you are logged in as a **non-root** user. You will install **PM2** global
 npm install pm2@latest -g
 ```
 
-Navigate to your **Strapi Project folder** and use the following command to set the environment variable to production and start `pm2`:
+### The ecosystem.config.js file
 
-`Path: ./my-project/`
+- You will need to configure an `ecosystem.config.js` file. This file will manage the **database connection variables** Strapi needs to connect to your database. The `ecosystem.config.js` will also be used by `pm2` to restart your project whenever any changes are made to files within the Strapi file system itself (such as when an update arrives from GitHub). You can read more about this file [here](https://pm2.io/doc/en/runtime/guide/development-tools/).
+
+- You will need to open your `nano` editor and then `copy/paste` the following:
 
 ```bash
-NODE_ENV=production pm2 start --name="strapi" server.js
+cd ~
+pm2 init
+sudo nano ecosystem.config.js
 ```
 
+- Next, replace the boilerplate content in the file, with the following:
+
+```js
+      module.exports = {
+  apps : [{
+    name: 'strapi',
+    cwd: '/home/path-to/strapi-project-folder'
+    script: 'server.js',
+    env: {
+      NODE_ENV: 'production',
+      DATABASE_HOST: 'localhost', // database endpoint
+      DATABASE_PORT: '5432',
+      DATABASE_NAME: 'strapi',  // DB name
+      DATABASE_USERNAME: 'your-name', // your username for psql
+      DATABASE_PASSWORD: 'password', // your password for psql
+    },
+  }],
+};
+
+```
+
+Use the following command to start `pm2`:
+
+```bash
+cd ~
+pm2 start ecosystem.config.js
+```
+
+`pm2` is now set-up to use an `ecosystem.config.js` to manage restarting your application upon changes. This is a recommended best practice.
```

Comment:
`pm2` is now set-up to use the `ecosystem.config.js` file to manage the restart if your application upon changes. This is a recommended best practice.

Reply:
I would seriously advise against promoting auto-restarting in production. That is a terrible thing to suggest or recommend. Just saying 😉

Reply:
@soupette - I left out the word "upon" as that seems like a typo and implemented your suggestion. 
@derrickmehaffy - I clarified this sentence about what we're recommending.  As it was unclear. I have heard differences of opinions about this point which tend to relate to the size of the project.  It seems evident that larger projects will pass through a staging process, and smaller projects managed by one person will have merges update the production site as a matter of intended action. For example, for my blog or a business website with one maintainer.  

But I definitely clarified it because we're only recommending using the ecosystem file to manage the restart.

Reply:
@davidkartuzinski even in the case of a single maintainer, Strapi has no database migration process. There is very real potential to seriously f*ck up your database structure if major changes are made and backups are something that absolutely should be recommended. Regardless of project size, recommending users to perform an action that has a high risk of damaging something is not a good idea. Leave it to the users to decided how to manage it.

Reply:
@derrickmehaffy yes yes, I made this clearer, and I see needing to make it explicit.  Thank you for further clarifying.